src/commands/test/lib/run-tests-command.ts

```diff
@@ -248,8 +267,32 @@ export abstract class RunTestsCommand extends AppCommand {
     return await uploader.uploadAndStart();
   }
 
+  protected generateReportPath(): string {
+    if (path.isAbsolute(this.testOutputDir)) {
+      return this.testOutputDir;
+    }
+    return path.join(process.cwd(), this.testOutputDir);
+  }
+
+  protected async mergeTestArtifacts(): Promise<void> {
+    // Each command should override it if needed
+  }
+
   private waitForCompletion(client: AppCenterClient, testRunId: string): Promise<number> {
     let checker = new StateChecker(client, testRunId, this.app.ownerName, this.app.appName, this.streamingOutput);
     return checker.checkUntilCompleted(this.timeoutSec);
   }
+
+  private async downloadArtifacts(testRunId: string, artifacts: TestArtifacts): Promise<void> {
+    for (var key in artifacts) {
+
+      var reportPath = this.generateReportPath();
+      fsHelper.createLongPath(reportPath);
+      await downloadUtil.downloadFileAndSave(artifacts[key], path.join(reportPath, `${key.toString()}.zip`));
+
+      this.streamingOutput.text((command: RunTestsCommand): string => {
+        return `##vso[task.setvariable variable=${key}]${testRunId}${os.EOL}`;
```

Comment:
Only just noticed this - the value should be the path to the .zip file (e.g. path.join(reportPath, `${key.toString()}.zip`))